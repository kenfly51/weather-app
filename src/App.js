import React, { useState } from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import "./App.scss";
import WeatherItem from "./Components/WeatherItem";
import SearchInput from "./Components/SearchInput";

const App = () => {
  const [foreCast, setForeCast] = useState([]);

  const renderWeatherItems = () => {
    return foreCast.slice(0, 5).map((item) => (
      <Col sm xs={12} key={item.id}>
        <WeatherItem item={item} />
      </Col>
    ));
  };

  return (
    <Container className="app">
      <Row>
        <Col sm={4} xs={12}>
          <SearchInput onSearchResult={setForeCast} />
        </Col>
      </Row>
      <Row>{renderWeatherItems()}</Row>
    </Container>
  );
};

export default App;
