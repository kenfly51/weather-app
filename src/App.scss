@import "~bootstrap/scss/bootstrap";

.app {
  $base-unit: 4px;

  .weather-item {
    text-align: center;
    align-items: center;
    margin-top: $base-unit * 4;

    .state-image {
      width: 70%;
      padding-top: $base-unit * 4;
    }

    .temp-details {
      margin-top: $base-unit * 4;
    }
  }
}
