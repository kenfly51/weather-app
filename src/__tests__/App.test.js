import { fireEvent, render } from "@testing-library/react";
import App from "../App";

jest.mock("../components/SearchInput", () => ({ onSearchResult }) => (
  <button
    data-testid="search-button"
    onClick={() =>
      onSearchResult([
        {
          id: Date.now,
          date: new Date("2021-01-26"),
          min: 11.2,
          max: 11.9,
          state: "state",
          stateAbbr: "ste",
        },
      ])
    }
  >
    Search
  </button>
));

describe("App", () => {
  it("should render empty forecast data if there is no data", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });

  it("should render forecast data if data available", () => {
    const { container, getByTestId } = render(<App />);
    const button = getByTestId("search-button");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});
