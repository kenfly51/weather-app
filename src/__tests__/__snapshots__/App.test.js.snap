// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render empty forecast data if there is no data 1`] = `
<div>
  <div
    class="app container"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-4 col-12"
      >
        <button
          data-testid="search-button"
        >
          Search
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        There is no data to display
      </div>
    </div>
  </div>
</div>
`;

exports[`App should render forecast data if data available 1`] = `
<div>
  <div
    class="app container"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-4 col-12"
      >
        <button
          data-testid="search-button"
        >
          Search
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm col-12"
      >
        <div
          class="weather-item card"
        >
          <img
            class="card-img-top state-image"
            src="https://www.metaweather.com/static/img/weather/ste.svg"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Tuesday
            </div>
            <div
              class="card-subtitle h6"
            >
              state
            </div>
            <div
              class="temp-details"
            >
              <div>
                min: 11
              </div>
              <div>
                max: 11
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
